pkg/cluster/config/v1alpha2/zz_generated.conversion.go

```diff
@@ -149,6 +123,38 @@ func Convert_config_LifecycleHook_To_v1alpha2_LifecycleHook(in *config.Lifecycle
 	return autoConvert_config_LifecycleHook_To_v1alpha2_LifecycleHook(in, out, s)
 }
 
+func autoConvert_v1alpha2_Node_To_config_Node(in *Node, out *config.Node, s conversion.Scope) error {
+	out.Replicas = (*int32)(unsafe.Pointer(in.Replicas))
+	out.Role = config.NodeRole(in.Role)
+	out.Image = in.Image
+	out.KubeadmConfigPatches = *(*[]string)(unsafe.Pointer(&in.KubeadmConfigPatches))
+	out.KubeadmConfigPatchesJSON6902 = *(*[]kustomize.PatchJSON6902)(unsafe.Pointer(&in.KubeadmConfigPatchesJSON6902))
+	out.ControlPlane = (*config.ControlPlane)(unsafe.Pointer(in.ControlPlane))
+	return nil
+}
+
+// Convert_v1alpha2_Node_To_config_Node is an autogenerated conversion function.
+func Convert_v1alpha2_Node_To_config_Node(in *Node, out *config.Node, s conversion.Scope) error {
+	return autoConvert_v1alpha2_Node_To_config_Node(in, out, s)
+}
+
+func autoConvert_config_Node_To_v1alpha2_Node(in *config.Node, out *Node, s conversion.Scope) error {
+	out.Replicas = (*int32)(unsafe.Pointer(in.Replicas))
+	out.Role = NodeRole(in.Role)
+	out.Image = in.Image
+	out.KubeadmConfigPatches = *(*[]string)(unsafe.Pointer(&in.KubeadmConfigPatches))
```

Comment:
ok thanks, I wonder why they're doing this so much.

Reply:
I can only guess:
- because conversion-gen always assumes that we are converting from TypeA.vx to TypeA.vY and does not detect cases when the type is external from the API (the type is the same)
or
- because this is a convenient way for doing "deepcopy" with types not supporting it explicitly